etl/generative_model_pipelines/open_ai_generative_model_pipeline.py

```diff
@@ -14,28 +14,35 @@
 )
 
 
-class OpenAiGenerativeModelResource(ConfigurableResource):  # type: ignore
-    """A ConfigurableResource to enrich Records using OpenAI's generative AI models."""
+class OpenAiGenerativeModelPipeline(GenerativeModelPipeline):
+    """
+    A concrete implementation of GenerativeModelPipeline.
 
-    openai_resource_params: OpenAiResourceParams
-    __template = """
+    Uses OpenAI's generative models to enrich Records.
+    """
+
+    def __init__(self, openai_pipeline_config: OpenAiPipelineConfig) -> None:
+        self.__openai_pipeline_config: OpenAiPipelineConfig = openai_pipeline_config
+        self.__template = """
                 Keep the answer as concise as possible.
                 Question: {question}
                 """
 
     def __create_question(self, record_key: RecordKey) -> ModelQuestion:
         """Return a question for an OpenAI model."""
 
-        match self.openai_resource_params.enrichment_type:
+        match self.__openai_pipeline_config.enrichment_type:
```

Comment:
Should have a default case here, probably one that throws. I'm surprised mypy let this pass.

Reply:
I noticed that too, but I thought a no default case was allowed for enums. 
I have added exceptions anyway. Just to be on the safer side.